DoSomething/gladiator
# Show competition users

#### What's this PR do?

Shows a table of users in a competition. 

I did a bunch of things in the PR -

* Hooked up to Phoenix so that we can pull user activity. I am holding off on showing actual signup activity for now because this was getting hard without any real phoenix users in the system. 
* Made a new endpoint `/users/{id}/signup/{signup_id}` that can show the user activity. TODO: add campaign run as a parameter so you can see only signup/reportback activity for the campaign and campaign run a user signed up for.
* Added a function in the Competitions model that gets all of the users in a competiton. I wasn't sure where the correct place to put a function like this was. Ideally, when we finish #45, there will be a Brackets table that can have a model that defines all the methods we need to work with brackets. For example, per competition, we might want to show all the brackets, and then under each bracket, all of the users in that bracket. 
* Added a seeder to create a dummy competition in my DB. I didn't add this to the root seeder class cause only wanted to run it, if I needed it, but it's there. 
* Added a `competition_users` seeded that put a random set of users in a competition so I can work with them. This will probably be depreciated when we move to the Bracket structure.
* Lastly, a small test that just checks that I am hitting and getting a response from phoenix. This is not a great test, but it was easier for me to do this then to do like a `dd()` and have to refresh my browser everytime. 


#### How should this be manually tested?
pull this branch, run the seeds, go to view a competition, and you should see: 

![screen shot 2016-03-04 at 2 20 22 pm](https://cloud.githubusercontent.com/assets/1700409/13538023/2319f958-e216-11e5-8456-d0288ad92e4b.png)


#### Any background context you want to provide?

This is still a work in progress, but gets us mostly there. 

There is more to be done re: showing user signup data, which can be done easier now since we have db seeds that have real phoenix users. 

The test needs to be fixed to use mock users.

#### What are the relevant tickets?

Address #27 

cc @weerd @angaither
